src/core/src/core_logic/PatchInstaller.py

```diff
@@ -202,52 +238,43 @@ def install_updates(self, maintenance_window, package_manager, simulate=False):
             self.composite_logger.log(progress_status)
 
             # include all dependencies (with specified versions) explicitly
+            # package_and_dependencies initially conains only one package. The dependencies are added in the list by method include_dependencies
             package_and_dependencies = [package]
             package_and_dependency_versions = [version]
-            dependencies = package_manager.get_dependent_list(package)
-            for dependency in dependencies:
-                if dependency not in all_packages:
-                    continue
-                package_and_dependencies.append(dependency)
-                package_and_dependency_versions.append(package_versions[packages.index(dependency)] if dependency in packages else Constants.DEFAULT_UNSPECIFIED_VALUE)
-
-            # multilib resolution for yum
-            if package_manager.get_package_manager_setting(Constants.PKG_MGR_SETTING_IDENTITY) == Constants.YUM:
-                package_name_without_arch = package_manager.get_product_name_without_arch(package)
-                for possible_arch_dependency, possible_arch_dependency_version in zip(packages, package_versions):
-                    if package_manager.get_product_name_without_arch(possible_arch_dependency) == package_name_without_arch and possible_arch_dependency not in package_and_dependencies:
-                        package_and_dependencies.append(possible_arch_dependency)
-                        package_and_dependency_versions.append(possible_arch_dependency_version)
-
-            # remove duplicates
-            package_and_dependencies, package_and_dependency_versions = package_manager.dedupe_update_packages(package_and_dependencies, package_and_dependency_versions)
+            
+            self.include_dependencies(package_manager, [package], all_packages, packages, package_versions, package_and_dependencies, package_and_dependency_versions)
 
```

Comment:
We still need #multilib yum resolution for these packages and removing duplicates right?

Reply:
multilib yum resolution is being done in the method include_dependencies through the new function add_arch_dependencies.

Reply:
What about removing duplicates?

Reply:
Removing duplicates in the include_dependencies method